sn_client/src/file_apis.rs

```diff
@@ -196,16 +202,17 @@ impl Files {
             let mut tasks = vec![];
             for chunk in next_batch {
                 let client = self.client.clone();
-                let closest_peers_clone = closest_peers.clone();
+                let all_peers_clone = all_peers.clone();
                 let chunk_addr = *chunk.address();
                 let payment = payment_proofs.get(&chunk_addr.name().0).cloned();
                 // TODO: re-enable requirement to always provide payment proof
                 //.ok_or(super::Error::MissingPaymentProof(chunk_addr))?;
 
                 tasks.push(task::spawn(async move {
-                    client
-                        .store_chunk(chunk, payment, closest_peers_clone)
-                        .await?;
+                    let dst = NetworkAddress::from_chunk_address(chunk_addr);
+                    let closest_peers =
+                        sort_peers_by_address(all_peers_clone, &dst, CLOSE_GROUP_SIZE)?;
+                    client.store_chunk(chunk, payment, closest_peers).await?;
```

Comment:
Same as the above one? Should we call the `local_closet_peers(chunk)` directly?

Reply:
No. by using `sort_peers_by_address`, there is then only one call from node to swarm_driver.
using `local_closet_peers(chunk)`, result in one call from node to swarm_driver for each chunk.
Though it is via `local event channel`, prefer to avoid such multiple calls.

Reply:
That is right! Looks good to me!